### ISSUE TITLE
Videos not in Order

### ISSUE THREAD

**Issue by ardalis** [DevBetterCom/DevBetterWeb#1005]
Just uploaded videos from 26th and 27th and they're appearing in this order:
![image](https://user-images.githubusercontent.com/782127/215149632-2d12a6dd-fe7a-488c-9b06-65ada11017ec.png)

Should be most recent first.

**Comment by samuelwine** [comment]
Could this be to do with the video DateCreated somehow set incorrectly? 
I tried to follow it back to see where DateCreated is set originally, but I couldn't quite find it.

**Comment by ardalis** [comment]
Very likely. I'm not sure where it is set, either, or if it's set by Vimeo and we just import their value. Something to track down for sure.

**Comment by totalitydevelopment** [comment]
The code grabs the full video list from the db, then grabs a cached list also.

Any videos that aren't in the cache get added at position 0. Guessing maybe given the close nature of those 2 videos recording time wise you uploaded the 27th before the 26th which left the 26th first in the list.

There is no ordering on the cached list being returned.

That would be the cheap and easy way to resolve...

**Comment by samuelwine** [comment]
@totalitydevelopment any chance you could add a link to the code you are referencing?

Thanks!

**Comment by totalitydevelopment** [comment]
@samuelwine try this:
https://github.com/DevBetterCom/DevBetterWeb/blob/main/src/DevBetterWeb.Web/Endpoints/VideoEndpoints/GetAllVideos.cs

**Comment by samuelwine** [comment]
thanks @totalitydevelopment.
I'm a bit unsure where that gets called - when i looked through the code, it looked like it was calling [this](https://github.dev/DevBetterCom/DevBetterWeb/blob/31aa1935755616cfd0f4f90b26951d5c44f0932b/src/DevBetterWeb.Web/Endpoints/VideoEndpoints/VideoListDataTable.cs#L36)

**Comment by totalitydevelopment** [comment]
Yeah you're right. Dived in at that code without running as only had a few minutes.

![image](https://user-images.githubusercontent.com/93525145/215440929-a5cf6316-4304-44f0-8f8b-f0debfdb9312.png)

**Comment by totalitydevelopment** [comment]
That'll need a manual db tweak for the outstanding issue... or you can could change the spec to order by dateUploaded but there will always be scope for them being uploaded in the wrong order that way...

**Comment by samuelwine** [comment]
@ardalis what is the process for creating a new video? I'm assuming it gets created locally, then gets uploaded to Vimeo, as well as an "overview" to the DB. Is that correct?
And does the Vimeo upload also get done through the DevBetter site?
Thanks

**Comment by ardalis** [comment]
There's a separate Uploaders github repo with a WPF app that does the upload to Vimeo. @ShadyNagy wrote it. It's interested the DateCreated is only Date, not time, which seems to be the issue.

**Comment by samuelwine** [comment]
Makes sense!
